subsys/bluetooth/host/audio/iso.c

```diff
@@ -256,32 +259,40 @@ void bt_iso_cleanup(struct bt_conn *conn)
 {
 	int i;
 
-	bt_conn_unref(conn->iso.acl);
-	conn->iso.acl = NULL;
+	if (!conn->iso.is_bis) {
+		bt_conn_unref(conn->iso.acl);
+		conn->iso.acl = NULL;
 
-	/* Check if conn is last of CIG */
-	for (i = 0; i < CONFIG_BT_MAX_ISO_CONN; i++) {
-		if (conn == &iso_conns[i]) {
-			continue;
-		}
+		/* Check if conn is last of CIG */
+		for (i = 0; i < CONFIG_BT_MAX_ISO_CONN; i++) {
+			if (conn == &iso_conns[i]) {
+				continue;
+			}
 
-		if (atomic_get(&iso_conns[i].ref) &&
-		    iso_conns[i].iso.cig_id == conn->iso.cig_id) {
-			break;
+			if (atomic_get(&iso_conns[i].ref) &&
+			    iso_conns[i].iso.cig_id == conn->iso.cig_id) {
+				break;
+			}
 		}
-	}
 
-	if (i == CONFIG_BT_MAX_ISO_CONN) {
-		hci_le_remove_cig(conn->iso.cig_id);
+		if (i == CONFIG_BT_MAX_ISO_CONN) {
+			hci_le_remove_cig(conn->iso.cig_id);
+		}
 	}
-
 	bt_conn_unref(conn);
-
 }
 
 struct bt_conn *iso_new(void)
 {
-	return bt_conn_new(iso_conns, ARRAY_SIZE(iso_conns));
+	struct bt_conn *iso = bt_conn_new(iso_conns, ARRAY_SIZE(iso_conns));
```

Comment:
In the current implementation (PR) of ISO in the Zephyr controller, we have separate channels for BIS and CIS, but in the host we have a single pool (`iso_conns`). I guess it would make sense to align these; what makes most sense? A shared pool, or separate pools? 

If we want to support switching from CIS to BIS, or vice versa, I suppose a shared pool makes most sense?